### PR TITLE
Add CODEOWNERS template and DCO signoff

### PR DESCRIPTION
**Pull request by jfriedri-ni**
### What does this Pull Request accomplish?

This PR updates NI's GitHub repo template to match the instructions in https://github.com/ni/.github/blob/main/profile/CONTRIBUTING.md

### Why should this Pull Request be merged?

The repo template needs to match NI's open source policies.

### What testing has been done?

None

### CODE REVIEW

**Review by benmont — APPROVED** [review]

**Comment by benmont** [comment]
@mbr0wn any concerns?

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (mbr0wn) [inline .github/PULL_REQUEST_TEMPLATE.md:15]: This seems inferior to enforcing people sign off commits using `Signed-off-by`. GitHub has apps to check that, like [this one](https://github.com/apps/dco).

Rationale: If we provide a checkbox here, it's something we need to manually/visually inspect. Sure, it's not hard. But it's a single-bit information, easier to miss than the DCO app's red checkmark.

If GitHub were to go away, or we lose access to the pull request, then we wouldn't have a written record of DCO approval by committers. If we enforce a signed-off-by, then it's in the repository.

And finally, it's making it a bit too easy for people to make PRs compliant. This is something where the little bit of additional friction is actually preferable to to potential legal issues later on, even if it's as "simple" as having to revert people's code at a later point in time.

**Review by mbr0wn — CHANGES_REQUESTED** [review]
See comment on DCO checkbox.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (benmont) [reply .github/PULL_REQUEST_TEMPLATE.md:15]: @mbr0wn @alejandro5042 should we remove this line from [Requirements for open source projects](https://github.com/ni/.github/blob/main/profile/CONTRIBUTING.md)?

“Include language in .github/pull_request_template.md that explicitly acknowledges the submission adheres to the DCO”

Comment (jfriedri-ni) [reply .github/PULL_REQUEST_TEMPLATE.md:15]: > This seems inferior to enforcing people sign off commits using `Signed-off-by`.

The [repo template](https://github.com/ni/github-repo-template/blob/8359f26efec5d2bd11d598add3c9b7caa7237110/CONTRIBUTING.md?plain=1#L12C1-L12C1) has this stronger requirement in its `CONTRIBUTING.md`:

> Please remember to sign off your commits (e.g., by using `git commit -s` if you
> are using the command line client). This amends your git commit message with a line
> of the form `Signed-off-by: Name Lastname <name.lastmail@emailaddress.com>`.

So I don't see how this a replacement or downgrade, but instead a both-and that meets the requirement that @benmont pointed out.

I browsed some pages in the repo settings, but I only found a toggle for _"Require contributors to sign off on web-based commits,"_ and none that required it for all commits. If there is a setting to highlight in [.github/profile/CONTRIBUTING.md](https://github.com/ni/.github/blob/main/profile/CONTRIBUTING.md#initial-setup), I can open a PR for that too.

Comment (mbr0wn) [reply .github/PULL_REQUEST_TEMPLATE.md:15]: Wait, those lines seem familiar...

image

Oh right, I put them there :smile: 

I agree with you that this is the less strong requirement. But that also means this is a subset of what we already have.

Instead of this checkbox, I will suggest that all new projects enable the DCO sign-off app. You are correct, there is no way to enforce signoffs when not using the GitHub page (because of how git works independently from GitHub). But I do think that this checkbox carries the risk of someone not checking git commits manually (which they would have to do without such a DCO signoff app). However, with the app, this checkbox becomes completely redundant.

@benmont Yeah thanks for pointing out that file. I think we should change that line to something stronger, e.g., "use an automated tool (e.g. <INSERT LINK>) to verify that contributors are using signed-off-by lines").

Comment (mbr0wn) [reply .github/PULL_REQUEST_TEMPLATE.md:15]: @jfriedri-ni @benmont What are your thoughts on using the DCO app? I could update the general CONTRIBUTING file.

Comment (jfriedri-ni) [reply .github/PULL_REQUEST_TEMPLATE.md:15]: > What are your thoughts on using the DCO app? I could update the general CONTRIBUTING file.

Seems like a good idea :)

Comment (mbr0wn) [reply .github/PULL_REQUEST_TEMPLATE.md:15]: I add this PR to the main repository: https://github.com/ni/.github/pull/1

@jfriedri-ni Can you please remove this section or replace it with something non-interactive (you must sign off commits... this will be checked automatically).

I'm OOO starting EOB tomorrow so I'll approve this and assume you can handle this. @benmont  I wasn't able to assign an owner for that PR mentioned above, can you merge it?

**Review by mbr0wn — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (benmont) [reply .github/PULL_REQUEST_TEMPLATE.md:15]: Yes I can merge after @jfriedri-ni removes/replaces with something.

**Comment by jfriedri-ni** [comment]
@benmont -- I do not have privileges to merge. Please complete this PR when you are able.

**Pull request merged**
